### ISSUE TITLE
[Bug]: AWS Bedrock: Error with OpenAI Client, working with direct Litellm API

### ISSUE THREAD

**Issue by lambda-science** [BerriAI/litellm#1981]
### What happened?

TL;DR: When using openAI client it is impossible to use the proxy with AWS Bedrock if the `AWS_REGION` env var is not set in LiteLLM container. And setting this env var override `litellm_config.yaml` models settings (impossible to have two regions for backup for example) !

Hey I have a weird error with LiteLLM Proxy server.
When I query the API directly it works perfectly:
```bash
curl --location 'http://localhost:8001/embeddings' \
--header 'Content-Type: application/json' \
--data '{
"model": "cohere-embed-docs-multilingual-v3",
"input": ["yeyeyeeyyeyeyeye"]
}'
```
But through openAI client it fails !
```python
import openai
client = openai.OpenAI(base_url="http://litellm:8001/", api_key="dummy")
client.embeddings.create(model="cohere-embed-query-multilingual-v3", input=["yeyeyeeyyeyeyeye"])
```
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.12/site-packages/openai/resources/embeddings.py", line 113, in create
    return self._post(
           ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/openai/_base_client.py", line 1200, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/openai/_base_client.py", line 889, in request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/openai/_base_client.py", line 980, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.AuthenticationError: Error code: 401 - {'error': {'message': 'BedrockException - BedrockException - AWS region not set: set AWS_REGION_NAME or AWS_REGION env variable or in .env file', 'type': None, 'param': None, 'code': 401}}
 ```
My litellm config is:
```yaml
model_list:
  - model_name: cohere-embed-query-multilingual-v3
    litellm_params:
      model: cohere.embed-multilingual-v3
      input_type: search_query
      aws_region_name: us-east-1
    model_info:
      mode: embedding
litellm_settings:
  set_verbose: False
  drop_params: True
```

Version `1.23.14`. My .aws folder with credential is mounted inside LiteLLM (and working as the request throught the Swagger interface works !) 
Also error 401 is "Forbidden", not internal error :pepe_think:
The solution for now is as said in the error message  `AWS region not set: set AWS_REGION_NAME or AWS_REGION env variable or in .env file` and it works. But I need it to work without it because I feel it creates more issues and confusion (during my complex AWS deployement) because of both AWS_REGION specified in the config file AND in env variable

The different behaviour between openAI client and direct API call is weird
Putting the environnement variable and calling it through OpenAI client override the config region !

If my LiteLLM Env Var is `eu-west-1` and LiteLLM model config is `us-east-1`:
- OpenAI client will use `eu-west-1` (LiteLLM's Env Var).
- While direct swagger API it uses `us-east-1` (LiteLLM's YAML config). Funny behaviour (and a bit problematic)

Setting additional `aws_region: us-east-1` (in addition to `aws_region_name`) in the `.yaml config` produce also and error with OpenAI Client:
```python
openai.BadRequestError: Error code: 400 - {'error': {'message': 'BedrockException - BedrockException - An error occurred (ValidationException) when calling the InvokeModel operation: Malformed input request: #: extraneous key [aws_region] is not permitted, please reformat your input and try again.', 'type': None, 'param': None, 'code': 400}}
```

**Comment by ishaan-jaff** [comment]
This is fixed on 1.40.27 @lambda-science 

<img width="775" alt="Screenshot 2024-06-26 at 1 25 26 PM" src="https://github.com/BerriAI/litellm/assets/29436595/a8b4ca98-ee2e-4307-b0da-7fb0a727cc39">

**Comment by ishaan-jaff** [comment]
@lambda-science any chance we can hop on a call ? I'd love to learn how how we can improve litellm for you + setup a direct support channel so we can resolve your issues fast 

I reached out to you on Linkedin if DMs work.
Sharing a link to my cal for your convenience: https://calendly.com/d/4mp-gd3-k5k/litellm-1-1-onboarding-chat
